jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceCollection.java

```diff
@@ -285,7 +285,10 @@ public List<Resource> list()
         List<Resource> result = new ArrayList<>();
         for (Resource r : _resources)
         {
-            result.addAll(r.list());
+            if (r.isDirectory())
+                result.addAll(r.list());
+            else
+                result.add(r);
```

Comment:
This is a strange extra case.  Do we ever have a use-case for a resource collection where it's members are not all directories?  Perhaps this is a restriction that can be put into ResourceCollection (not in this PR), so if we intend to do that, this is not required.

Reply:
Yes, a testcase has that example (some entries are dirs, some are files).
But a general review of this is a good idea.